dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/java_gateway.py

```diff
@@ -75,6 +78,21 @@ def __init__(
         auto_convert: Optional[bool] = True,
     ):
         self.java_gateway = launch_gateway(address, port, auto_convert)
+        gateway_version = self.get_gateway_version()
+        if gateway_version is not None and gateway_version != __version__:
+            logger.critical(
+                f"Using unmatched version of pydolphinscheduler (version {__version__}) "
+                f"and Java gateway (version {gateway_version}) may cause errors. "
+                "We strongly recommend you to find the matched version "
+                "(check: https://pypi.org/project/apache-dolphinscheduler)"
+            )
+
+    def get_gateway_version(self):
+        """Get the java gateway version, expected to be equal with pydolphinscheduler."""
+        try:
+            return self.java_gateway.entry_point.getGatewayVersion()
+        except Exception:
+            return None
```

Comment:
I do not think return `None` is a god idea, maybe we should raise error directly, WDTY

```py
    def get_gateway_version(self):
         """Get the java gateway version, expected to be equal with pydolphinscheduler."""
         return self.java_gateway.entry_point.getGatewayVersion()
```

Reply:
Thanks for the suggestion, the `try` clause is moved to __init__ cuz I think if there's a connection error or so, it could be more reasonable for user to get the error context from the method it intended to call.